Fix common typos and misspellings

SeaOtocinclus opened the pull request:
Summary:
This commit fixes numerous typos and misspellings found throughout the codebase. The fixes improve code readability and documentation consistency across C++, Python, CUDA, and documentation files.

## Typos Fixed

| Before | After | Occurrences |
|--------|-------|-------------|
| occured | occurred | 14 |
| accross | across | 9 |
| lenght/lenghts | length/lengths | 8 |
| unneccessary | unnecessary | 5 |
| Peform | Perform | 4 |
| furture | future | 3 |
| paritioned | partitioned | 2 |
| desireable | desirable | 2 |
| registerations | registrations | 2 |
| seperated | separated | 2 |
| intialized | initialized | 2 |
| capatibility | compatibility | 2 |
| peformed | performed | 2 |
| Exmple | Example | 2 |
| comma_seperated | comma_separated | 2 |
| cumsuming | consuming | 2 |
| neccessary | necessary | 1 |
| ParamterMetadataTable | ParameterMetadataTable | 1 |
| matached | matched | 1 |
| conaitner | container | 1 |
| reivew | review | 1 |
| prioriry | priority | 1 |
| Alocated | Allocated | 1 |
| opportunixtically | opportunistically | 1 |
| peformance | performance | 1 |
| equavalent | equivalent | 1 |
| asssumed | assumed | 1 |
| valdiation | validation | 1 |
| apprear | appear | 1 |
| consectuve | consecutive | 1 |
| dependending | depending | 1 |
| copnversion | conversion | 1 |
| weigted | weighted | 1 |
| repreesenting | representing | 1 |
| finialize | finalize | 1 |
| unintialized | uninitialized | 1 |
| conbined | combined | 1 |
| tesnor | tensor | 1 |
| desugared | discarded | 1 |
| behaviour | behavior | 1 |
| paramerizaitons | parametrizations | 1 |
| compute_output_lenghths_kernel | compute_output_lengths_kernel | 1 |

Test Plan: N/A - mostly comments - waiting on CI

Differential Revision: D83695665




cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168

SeaOtocinclus commented:
@eqy  can you take a look to this PR?

eqy commented on torch/export/exported_program.py:
```diff
@@ -532,16 +532,16 @@ def _is_joint_ir_decomp(ep, joint_loss_index):
         # and update the ep.state_dict. Ideally, we should just return
         # the state dict of ep.module but ep.module only stores params
-        # buffers that participate in forward. If we undo this behaviour,
+        # buffers that participate in forward. If we undo this behavior,
```
also not really a typo

eqy reviewed (changes requested):
I don't think the table in the original post is in sync with the changes? Also see some nits on incorrect "corrections"

SeaOtocinclus commented on torch/export/exported_program.py:
```diff
@@ -532,16 +532,16 @@ def _is_joint_ir_decomp(ep, joint_loss_index):
+        # When we apply parameterization rule to unwrap
         # subclasses, the state dict will now have different
-        # desugared parameters. We need to manually filter those
+        # disregarded parameters. We need to manually filter those
```
Got it, I will put that back.

SeaOtocinclus commented:
I will update the DIFF summary, with only the change relevant to the PR

SeaOtocinclus committed:
Fix common typos and misspellings (#164413)

Summary:

This commit fixes numerous typos and misspellings found throughout the codebase.

## Typos Fixed

| Before | After | Occurrences |
|--------|-------|-------------|
| Peform | Perform | 4 |
| future | future | 3 |
|lenghths| lengths| 1|
|occured|occurred| 1|
|parameterixzation|parameterization| 1|
|paramerizaitons|parameterizations| 1|
|unneccessary|unnecessary| 1|

Test Plan: N/A - mostly comments - waiting on CI

Differential Revision: D83695665

larryliu0820 reviewed (approved):
Review automatically exported from Phabricator review in Meta.

SeaOtocinclus commented:
@pytorchbot merge
